spec/koans.js

```diff
@@ -382,83 +394,92 @@ describe("destructuring function parameters. ", () => {
+    const [, ...butTheFirst] = [1, 2, 3, 4];
+    expect(butTheFirst).toEqual([2, 3, 4]);
   });
 
   it("extracts each array item", function () {
-    const [] = [...[1, 2]];
-    // expect(a).toEqual(1);
-    // expect(b).toEqual(2);
+    const [a, b] = [...[1, 2]];
+    expect(a).toEqual(1);
+    expect(b).toEqual(2);
   });
 
   it("in combination with rest", function () {
-    const [a, b, ...theRest] = [...[0, 1, 2, 3, 4, 5]];
-    // expect(a).toEqual(1);
-    // expect(b).toEqual(2);
-    // expect(theRest).toEqual([3, 4, 5]);
+    const [, a, b, ...theRest] = [...[0, 1, 2, 3, 4, 5]];
+    expect(a).toEqual(1);
+    expect(b).toEqual(2);
+    expect(theRest).toEqual([3, 4, 5]);
   });
 
   it("spreading into the rest", function () {
-    const [...rest] = [...[, 1, 2, 3, 4, 5]];
-    // expect(rest).toEqual([1, 2, 3, 4, 5]);
+    const [, ...rest] = [...[, 1, 2, 3, 4, 5]];
+    expect(rest).toEqual([1, 2, 3, 4, 5]);
   });
 });
 
 describe("spread with strings", () => {
   it("simply spread each char of a string", function () {
-    const [b, a] = ["xy"];
-    // expect(a).toEqual("y");
-    // expect(b).toEqual("x");
+    const [b, a] = [..."xy"];
+    expect(a).toEqual("y");
+    expect(b).toEqual("x");
   });
 
   it("works anywhere inside an array (does not have to be last)", function () {
-    const letters = ["a", "bcd", "e", "f"];
-    // expect(letters.length).toEqual(6);
+    const letters = ["a", ..."bcd", "e", "f"];
+    expect(letters.length).toEqual(6);
   });
 });
 
 describe("class creation", () => {
   it("is as simple as `class XXX {}`", function () {
-    let TestClass = {};
+    class TestClass {}
 
-    // const instance = new TestClass();
-    // expect(typeof instance).toBe('object');
+    const instance = new TestClass();
+    expect(typeof instance).toBe("object");
   });
 
   it("special method is `constructor`", function () {
     class User {
-      constructor(id) { }
+      constructor(id) {
+        this.id = id;
+      }
     }
 
     const user = new User(42);
-    // expect(user.id).toEqual(42);
+    expect(user.id).toEqual(42);
   });
 
   it("defining a method is simple", function () {
-    class User { }
+    class User {
+      writesTests() {
+        return false;
+      }
+    }
 
     const notATester = new User();
-    // expect(notATester.writesTests()).toBe(false);
+    expect(notATester.writesTests()).toBe(false);
   });
 
   it("multiple methods need no commas (opposed to object notation)", function () {
     class User {
       wroteATest() {
         this.everWroteATest = true;
       }
-      isLazy() { }
+      isLazy() {
+        if (this.wroteATest() === true) return false;
+        else return true;
+      }
     }
 
     const tester = new User();
-    // expect(tester.isLazy()).toBe(true);
+    expect(tester.isLazy()).toBe(true);
     tester.wroteATest();
-    // expect(tester.isLazy()).toBe(false);
+    // expect(tester.isLazy()).toBe(false); don't know how to deal with it
```

Comment:
i understand the confusion. in this case you should have rewriten the class and the methods a little.
the user, would have isLazy by default set to true, and whenever the method wroteATest() was called, the isLazy would be set to false.

Something like this maybe
```
    class User {
      constructor() {
        this.lazy = true;
      }
      wroteATest() {
        this.everWroteATest = true;
        this.lazy = false;
      }
      isLazy() {
        return this.lazy;
      }
    }

    const tester = new User();
    expect(tester.isLazy()).toBe(true);
    tester.wroteATest();
    expect(tester.isLazy()).toBe(false)
```